core/src/main/java/org/sonarsource/sonarlint/core/AbstractSonarLintEngine.java

```diff
@@ -46,4 +53,43 @@ public void fireModuleFileEvent(Object moduleKey, ClientModuleFileEvent event) {
       moduleContainer.getComponentByType(ModuleFileEventNotifier.class).fireModuleFileEvent(event);
     }
   }
+
+  protected <T> T withModule(AbstractAnalysisConfiguration configuration, Function<ComponentContainer, T> consumer) {
+    boolean deleteModuleAfterAnalysis = false;
+    Object moduleKey = configuration.moduleKey();
+    ComponentContainer moduleContainer = getModuleRegistry().getContainerFor(moduleKey);
+    if (moduleContainer == null) {
+      // if not found, means we are outside of any module (e.g. single file analysis on VSCode)
+      moduleContainer = getModuleRegistry().createContainer(new ModuleInfo(moduleKey, new AnalysisScopeFileSystem(configuration.inputFiles())));
+      deleteModuleAfterAnalysis = true;
+    }
+    try {
+      return consumer.apply(moduleContainer);
+    } finally {
+      if (deleteModuleAfterAnalysis) {
+        moduleContainer.stopComponents();
+      }
+    }
+  }
+
+  private static class AnalysisScopeFileSystem implements ClientFileSystem {
+
+    private final Iterable<ClientInputFile> filesToAnalyze;
+
+    private AnalysisScopeFileSystem(Iterable<ClientInputFile> filesToAnalyze) {
+      this.filesToAnalyze = filesToAnalyze;
+    }
+
+    @Override
+    public Stream<ClientInputFile> files(String suffix, InputFile.Type type) {
+      return files()
+        .filter(file -> file.relativePath().endsWith(suffix))
```

Comment:
I think it is enough to use filename instead of relativePath

Reply:
I didn't find this on ClientInputFile

Reply:
My bad, this is on InputFile